hazelcast/src/main/java/com/hazelcast/config/ConsistencyCheckStrategy.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Strategy for checking the consistency of data between replicas.
+ *
+ * @since 3.11
+ */
+public enum ConsistencyCheckStrategy {
+    /**
+     * No consistency checks
+     */
+    NONE((byte) 0),
+    /**
+     * Uses merkle trees (if configured) for consistency checks.
```

Comment:
are we failing fast when merkle tree is not configured but `ConsistencyCheckStrategy` is chosen as `MERKLE_TREES`?

Reply:
We can. This PR is just the config classes.